src/rechunk.jl

```diff
@@ -0,0 +1,16 @@
+# Force disk any abstractarray into a different chunking pattern.
+# This is useful in `zip` and other operations that can iterate
+# over multiple arrays with different patterns.
+
+struct RechunkedDiskArray{T,N,A<:AbstractArray{T,N},C} <: AbstractDiskArray{T,N}
```

Comment:
I really support adding this struct, it would have been useful in many occasions, and we might also re-formulate the basic broadcast using this.

Reply:
I had to look around to make sure we didn't have it already, it seems like such a basic component. But I hadn't thought about it either before zip.